java-backend/src/main/java/com/project/spring/DTO/Employee.java

```diff
@@ -0,0 +1,95 @@
+package com.project.spring.DTO;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotBlank;
+import java.util.Date;
+
+@Entity
+@Table(name = "employees")
+@JsonIgnoreProperties
+public class Employee implements Comparable <Employee>{
+
+    @Column(name = "emp_no")
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @NotBlank
+    private int empNo;
+
+    @Column(name = "birth_date")
+    @NotBlank
+    private String birthDate;
+
+    @Column(name = "first_name")
+    @NotBlank
+    private String firstName;
+
+    @Column(name = "last_name")
+    @NotBlank
+    private String lastName;
+
+    @Column(name = "gender")
+    @NotBlank
+    private char sex;
+
+    @Column(name = "hire_date")
+    @NotBlank
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
+    private Date hireDate;
+
+    @Override
+    public int compareTo(Employee e) {
+        return this.getHireDate().compareTo(e.getHireDate());
```

Comment:
This might not be the best way to utilize a sort for your Employees, because by putting this comparator within the Employees class, you're saying this is the canon definition of natural ordering.